src/libs/Navigation/AppNavigator/MainDrawerNavigator.js

```diff
@@ -1,49 +1,104 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+import _ from 'underscore';
 import {createDrawerNavigator} from '@react-navigation/drawer';
+import {withOnyx} from 'react-native-onyx';
 
 import withWindowDimensions, {windowDimensionsPropTypes} from '../../../components/withWindowDimensions';
+import FullScreenLoadingIndicator from '../../../components/FullscreenLoadingIndicator';
 import styles, {
     getNavigationDrawerType,
     getNavigationDrawerStyle,
 } from '../../../styles/styles';
+import ONYXKEYS from '../../../ONYXKEYS';
+import compose from '../../compose';
+import SCREENS from '../../../SCREENS';
 
 // Screens
 import SidebarScreen from '../../../pages/home/sidebar/SidebarScreen';
 import ReportScreen from '../../../pages/home/ReportScreen';
 
 const propTypes = {
+    // Initial report to be used if nothing else is specified by routing
+    initialReportID: PropTypes.string,
+
     ...windowDimensionsPropTypes,
 };
 
+const defaultProps = {
+    initialReportID: null,
+};
+
 const Drawer = createDrawerNavigator();
+const LoadingScreen = React.memo(() => <FullScreenLoadingIndicator visible />, () => true);
+
+class MainDrawerNavigator extends React.Component {
+    constructor(props) {
+        super(props);
+        this.shouldMountReportScreen = _.isString(props.initialReportID);
```

Comment:
It might not be clear that an `initialReportID` should always either exist or be on its way soon. Could we document this somehow?